### PR TITLE
Add support for using symbols instead of strings or dot tags.

### PR DESCRIPTION
**Pull request by wildwestrom**
```clojure
;; You can now write
[:tag {:class '[class-name another-class-name]} "content"]
;; As opposed to just
[:tag {:class '["class-name another-class-name"]} "content"]
;; Or
[:tag.class-name.another-class-name "content"]
```

There's also some refactoring that went into this. Feel free to cherry pick just from lines 61-62 in `processor.clj`.

### CODE REVIEW

**Comment by green-coder** [comment]
Hi @wildwestrom , thank you for the contribution.

As it is now, it is hard for me to accept the PR and just press the green "Merge" button because it contains a lot more changes than just "adding the gathering of the symbols to the preprocessor".

Please avoid refactorings unless it solves a real problem. The same for the re-indentations, unless it is to fix an inconsistency in the code base.

At last, please re-group your commits together.

File: example/reagent-demo/src/acme/frontend/drop_in.cljc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+(ns acme.frontend.drop-in
+  (:require [girouette.core :refer [css]]))
+
+;; This proves that girouette
+;; can work as a drop-in replacement for tailwind.
```
Comment (green-coder) [inline example/reagent-demo/src/acme/frontend/drop_in.cljc:5]: Tailwind is a JS library and JS does not have symbols.

Please explain exactly how it is a drop-in replacement for Tailwind, it is not clear to me.

File: lib/processor/.gitignore [hunk 2]
```diff
@@ -6,3 +6,7 @@
 .cljs_node_repl
 node_modules
 out
+
+*~
+[#]*[#]
+.\#*
```
Comment (green-coder) [inline lib/processor/.gitignore:12]: I merged it in a separate commit, you can remove this from this PR.

File: lib/processor/deps.edn [hunk 3]
```diff
@@ -9,29 +9,29 @@
 
         garden/garden {:mvn/version "1.3.10"}
         ;girouette/girouette {:local/root "../girouette"}
-        girouette/girouette {:mvn/version "0.0.2"}}
+        girouette/girouette {:mvn/version "0.0.3"}}
```
Comment (green-coder) [inline lib/processor/deps.edn:12]: It works without changing the version. The version used in the user's project will override it.

File: lib/processor/deps.edn [hunk 4]
```diff
@@ -9,29 +9,29 @@
...
-                     :exec-args {:sync-pom true
-                                 :group-id "girouette"
-                                 :artifact-id "processor"
-                                 :version "0.0.2"
-                                 :jar "processor.jar"}}}}
+                     :exec-fn      hf.depstar/jar
+                     :exec-args    {:sync-pom    true
+                                    :group-id    "girouette"
+                                    :artifact-id "processor"
+                                    :version     "0.0.3"
```
Comment (green-coder) [inline lib/processor/deps.edn:33]: This is the version for the release tool. I will change it when it is time to make a release of the processor.

File: lib/processor/src/girouette/processor.clj [hunk 5]
```diff
@@ -15,65 +15,35 @@
...
         stubbed-js-deps (zipmap (concat npm-deps foreign-libs)
                                 (repeatedly #(gensym "fake$module")))]
     (swap! state update :js-dependency-index #(merge stubbed-js-deps %))))
 
-
 (defonce state
   (let [s (ana-api/empty-state)]
     (ana-api/with-state s
-     (comp/with-core-cljs))
+      (comp/with-core-cljs))
```
Comment (green-coder) [inline lib/processor/src/girouette/processor.clj:41]: ok for this change

File: lib/processor/src/girouette/processor.clj [hunk 6]
```diff
@@ -82,29 +52,46 @@
...
     ast))
 
+(def receive-hook-fn
+  {:keyword #(->> (name %)
+                  (re-seq #"\.[^\.#]+")
+                  (map (fn [s] (subs s 1))))
+   :string  #(->> (str/split % #" ")
+                  (remove str/blank?))
+   :symbol  #(->> (name %)
+                  (re-seq #"[\S]+"))})
```
Comment (green-coder) [inline lib/processor/src/girouette/processor.clj:62]: I am not sure to understand why we need reg-ex here.

**Comment by wildwestrom** [comment]
Ok, let me rescind this and just give the essential change.

**Pull request closed without merging**